runtime/src/lib.rs

```diff
@@ -1,6 +1,27 @@
 use std::ffi::c_void;
 use std::{fmt, ptr};
 
+#[no_mangle]
```

Comment:
nit: attributes at the bottom

(they're harder to notice above the docs, especially if the docs are long)